gammapy/makers/background/reflected.py

```diff
@@ -451,6 +451,13 @@ def __init__(
         if exclusion_mask and not exclusion_mask.is_mask:
             raise ValueError("Exclusion mask must contain boolean values")
 
+        if exclusion_mask and not exclusion_mask.geom.is_flat:
+            raise ValueError("Exclusion mask must only contain spatial dimension")
+
+        if exclusion_mask:
+            exclusion_mask = exclusion_mask.sum_over_axes(keepdims=False)
```

Comment:
To summarize,
- if the non-spatial axis has a dimension equal to 1 (ie is_flat), then there is an error
- if the non-spatial axis dimension is strictly larger than, then we sum over the axes.
Is it correct?

With event_type, then, it should also work in principle...

Reply:
This will accept only 1 bin long non spatial axes. If we want to support multiple label axes for datasets with label axes, we will have to introduce the feature in general. But in another PR, once `Makers` support additional axes.